src/Compilers/Core/CommandLine/BuildClient.cs

```diff
@@ -149,7 +166,11 @@ public Task<RunCompilationResult> RunCompilationAsync(IEnumerable<string> origin
 
         protected abstract int RunLocalCompilation(string[] arguments, BuildPaths buildPaths, TextWriter textWriter);
 
-        protected abstract Task<BuildResponse> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string sessionName, string keepAlive, string libDirectory, CancellationToken cancellationToken);
+        /// <summary>
+        /// Attempts to run the compilation on the server.  This function should never return a Task in a faulted
+        /// state.  The task should only return in a completed or cancelled state.
```

Comment:
It doesn't look like we rely on any async behavior here, why return a Task at all?